### PR TITLE
[receiver/hostmetrics] Migrate Processes scraper to the Metrics builder

### LINKED ISSUE #7140: Migrate hostmetrics / processesscraper to the new Metrics Builder
receiver/hostmetricsreceiver/internal/scraper/processesscraper

### PR DESCRIPTION
**Pull request by dmitryax**
This change migrates the last hostmetrics receiver scraper to the new Metrics builder.

This also now allows more the scraper configurations currently provided by the metrics builder:
- Enabling/disabling particular metrics

Closes: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/7140

### CODE REVIEW

**Comment by dmitryax** [comment]
cc @codeboten @djaglowski

**Review by djaglowski — APPROVED** [review]

File: receiver/hostmetricsreceiver/internal/scraper/processesscraper/internal/metadata/generated_metrics_v2.go [hunk 1]
```diff
@@ -0,0 +1,274 @@
+// Code generated by mdatagen. DO NOT EDIT.
```
Comment (dashpole) [inline receiver/hostmetricsreceiver/internal/scraper/processesscraper/internal/metadata/generated_metrics_v2.go:1]: why rename to generated_metrics_v2.go?

Comment (dmitryax) [reply receiver/hostmetricsreceiver/internal/scraper/processesscraper/internal/metadata/generated_metrics_v2.go:1]: We chose this filename to differentiate between the old and new generated code. `cmd/mdatagen` uses it to validate if the code is generated. One all the scrapes migrated, we will remove the suffix.

Comment (dashpole) [reply receiver/hostmetricsreceiver/internal/scraper/processesscraper/internal/metadata/generated_metrics_v2.go:1]: ah, got it.

**Review by dashpole — APPROVED** [review]

**Comment by jpkrohling** [comment]
Looks like there's a conflict now.

**Comment by dmitryax** [comment]
@jpkrohling , rebased

**Review by codeboten — APPROVED** [review]

**Comment by jpkrohling** [comment]
I just rebased and moved the changelog entry to unreleased. This is ready to be merged.

**Comment by dmitryax** [comment]
@codeboten @jpkrohling can you please help with merging this?

**Review by codeboten — APPROVED** [review]

**Pull request merged**